src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -995,7 +995,13 @@ static ImmutableArray<TNamespaceOrTypeSymbol> createMembers(object value)
             where TNamespaceOrTypeSymbol : class
             where TNamedTypeSymbol : class, TNamespaceOrTypeSymbol
         {
-            var dictionary = new Dictionary<TKey, ImmutableArray<TNamedTypeSymbol>>(comparer);
+            // Initialize dictionary capacity to avoid resize allocations during Add calls.
+            // Most iterations through the loop add an entry. If map is smaller than the
+            // smallest capacity dictionary will use, we'll let it grow organically as
+            // it's possible we might not add anything to the dictionary.
+            var capacity = map.Count > 3 ? map.Count : 0;
```

Comment:
>var capacity = map.Count > 3 ? map.Count : 0; [](http://example.com/codeflow?start=12&length=45)

I guess we could iterate through the "map" and get an accurate count

Reply:
This estimate felt "good enough". Getting the actual count essentially requires doing most of the work from getOrCreateNamedTypes on each item in the map, which would probably make this perform worse than originally.

Reply:
> This estimate felt "good enough". 

No objection then